source/common/network/matching/data_impl.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "envoy/network/filter.h"
+
+namespace Envoy {
+namespace Network {
+namespace Matching {
+
+/**
+ * Implementation of Network::MatchingData, providing connection-level data to
+ * the match tree.
+ */
+class MatchingDataImpl : public MatchingData {
+public:
+  MatchingDataImpl(const ConnectionSocket& socket) : socket_(socket) {}
+  static absl::string_view name() { return "network"; }
```

Comment:
I don't think you need this on the child impl, as registration is done against MatchingData

Reply:
Done, removed.